### ISSUE TITLE
Missing state messages.

### ISSUE THREAD

**Issue by mrekai** [Exa-Networks/exabgp#325]
exabgp 3.4.12 on FreeBSD will sometimes pass JSON state messages via the API, other times it will not.  On one run,a process will receive state messages of type state with neighbor state of up, down, connected.  After a stop and restart of the exabgp process, perhaps of these state messages will be received.  Restart again, maybe they will show up -- varies from run to run, seems completely non-deterministic.  This makes it difficult for a process to remove installed routes when not receiving down notifications about a neighbor.

Messages of interests:
```
{
    "counter": 1,
    "exabgp": "3.4.8",
    "host": "some.host",
    "neighbor": {
        "address": {
            "local": "fec0::1:1",
            "peer": "fec0::1:2"
        },
        "asn": {
            "local": "65003",
            "peer": "65004"
        },
        "ip": fec0::1:2",
        "state": "connected"
    },
    "pid": "36591",
    "ppid": "1",
    "time": 1440603404.16,
    "type": "state"
}
```
And also those where neighbor "state": "up" and "state": "down".

**Comment by thomas-mangin** [comment]
Hello @mrekai - this is very little information to go by with. Could you please provide more information as AFAIK and can see the code behaves correctly. Could you provide what your code looks like ( if required sending me a private mail ) ?

**Comment by mrekai** [comment]
I first considered the possibility that my code was eating the messages.  I been able to record the absense of state messages with a very simple test case of code:

#!/bin/sh

while read line
do
  echo $line >> /tmp/exarecv
done

Relevant exabgp config:

  process writer {
    encoder json;
    receive {
      parsed;
      update;
      neighbor-changes;
      notification;
    }
    run /usr/local/etc/exabgp/writer;
  }

And on each of the neighbors:

    process {
      receive {
        parsed;
        update;
        neighbor-changes;
        notification;
      }
    }

**Comment by thomas-mangin** [comment]
@mrekai it is possible that the message is send IF the session is established by ExaBGP but missed if the session is established by the remote end. I will check this possibility and come back here with more information when I done.

**Comment by mrekai** [comment]
I had considered this possibility, but haven't found a good way to verify who establishes or to force one to do so instead of the other.  If there is a method, please advise and I will try.  Also now running 3.4.12 from Git.

**Comment by thomas-mangin** [comment]
You can use the option "passive" on a neighbor to prevent it to make ongoing connections.

**Comment by mrekai** [comment]
Hrm, this seems to make the Ciscos I'm using in this case never come up.  They go idle, active, open sent, idle.  Not going to worry about that now.

**Comment by thomas-mangin** [comment]
Could you please run ExaBGP with the -d option so I can see the reason why it does not work ?

**Comment by thomas-mangin** [comment]
Hello - could you please check if the latest 3.4 branch is still broken or not please, following the fix made for #337

**Comment by mrekai** [comment]
This behavior looks to have corrected with the fix for #337.  Awesome.

**Comment by thomas-mangin** [comment]
That is a relief 😄 thank you for letting me know.
